rust/agama-locale-data/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+use serde::Deserialize;
+use std::fs::File;
+use std::io::BufReader;
+use quick_xml::de::Deserializer;
+use flate2::bufread::GzDecoder;
+
+pub mod keyboard;
+pub mod language;
+pub mod territory;
+pub mod timezone_part;
+pub mod ranked;
+
+pub fn get_keyboards() -> keyboard::Keyboards {
+    const FILE_PATH: &str = "/usr/share/langtable/data/keyboards.xml.gz";
+    let file = File::open(FILE_PATH).expect("Failed to read langtable-data.");
+    let reader = BufReader::new(GzDecoder::new(BufReader::new(&file)));
+    let mut deserializer = Deserializer::from_reader(reader);
+    keyboard::Keyboards::deserialize(&mut deserializer).expect("Failed to deserialize keyboard entry")
+}
+
+pub fn get_languages() -> language::Languages {
+    const FILE_PATH: &str = "/usr/share/langtable/data/languages.xml.gz";
+    let file = File::open(FILE_PATH).expect("Failed to read langtable-data.");
```

Comment:
All the functions look similar. I guess we could reduce the repetition by extracting some logic to a separate function.